src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs

```diff
@@ -227,7 +227,16 @@ internal void InvokeSelectionChanged(object[] removedItems, object[] addedItems)
 
 		public int SelectedIndex
 		{
-			get => (int)this.GetValue(SelectedIndexProperty);
+			get
+			{
+				var value = (int)this.GetValue(SelectedIndexProperty);
```

Comment:
This isn't the correct fix - it won't have any effect if SelectedIndex is bound in the view model for example. 

WinUI has some rather complicated logic for 'remembering' the SelectedIndex or SelectedItem if they're set before items are available, and fixing this bug will involve importing that code.

Reply:
@davidjohnoliver Can you point me to where this logic exists?

Reply:
It's closed-source currently, this issue will probably need to be tackled by I or someone else at nventive who has access.

Reply:
We can merge the tests though, with `[Ignore]` and a TODO linking them to #4686.

Reply:
➡️ https://github.com/unoplatform/uno/pull/6786